pom.xml

```diff
@@ -1848,5 +1848,37 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>ci</id>
+            <properties>
+                <!-- the *local* master, not refs/remotes/... -->
+                <gib.referenceBranch>master</gib.referenceBranch>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.vackosar.gitflowincrementalbuilder</groupId>
+                        <artifactId>gitflow-incremental-builder</artifactId>
+                        <version>3.14.5</version>
+                        <extensions>true</extensions>
+                        <configuration>
+                            <disableIfBranchMatches>master</disableIfBranchMatches>
+                            <compareToMergeBase>true</compareToMergeBase>
+                            <uncommitted>true</uncommitted>
+                            <untracked>true</untracked>
+                            <excludePathsMatching>(\.github([/\\].*)?|\.mvn([/\\].*)?|mvnw|README.md|LICENSE)</excludePathsMatching>
+                            <buildDownstream>true</buildDownstream>
+                            <buildUpstream>true</buildUpstream>
+                            <buildUpstreamMode>impacted</buildUpstreamMode>
+                            <skipTestsForUpstreamModules>true</skipTestsForUpstreamModules>
+                            <argsForUpstreamModules>-Dmaven.site.skip=true -Dmaven.javadoc.skip=true -Dair.check.skip-all</argsForUpstreamModules>
```

Comment:
Comment added. GIB builds upstream modules since they're dependencies of the modified modules and are required to be built. It does not skip any checks in modified modules.

Reply:
I started understanding more after taking this branch for a spin locally. For other readers GIB only deals with compilation of modules and the `upstream` modules refer to what maven would build when passed `--also-make`.

This does raise another question - sorry if it's stupid:

Skipping checks is safe because if there was a change in an upstream module it would be part of the build anyway and the `argsForUpstreamModules` won't apply to it?

e.g. If I change `trino-testing-services` and a dependent module like `trino-postgresql` but I introduce a checkstyle failure in `trino-testing-services` (upstream of `trino-postgresql`) then would it get ignored due to `argsForUpstreamModules`?

Reply:
It's a good question - yes, this is how I understand this, that the tree of dependencies (upstream modules) is separate from the list of impacted (modified) modules.

But your example is not correct. If `trino-testing-services` is impacted (modified), then it's built with the flags you pass to Maven directly, not `argsForUpstreamModules`.

The GIB docs are pretty good, but you have to go through most options since they're related: https://github.com/gitflow-incremental-builder/gitflow-incremental-builder/#gibargsforupstreammodules